opensafely/check.py

```diff
@@ -97,12 +101,14 @@ def get_repository_name():
     if "GITHUB_REPOSITORY" in os.environ:
         return os.environ["GITHUB_REPOSITORY"]
     else:
-
-        url = subprocess.run(
-            args=["git", "config", "--get", "remote.origin.url"],
-            capture_output=True,
-            text=True,
-        ).stdout
+        git_config_path = Path(".git", "config")
+        if not git_config_path.is_file():
+            return
+        config = configparser.ConfigParser()
+        config.read(git_config_path)
+        if 'remote "origin"' not in config.sections():
+            return
+        url = config['remote "origin"']["url"]
```

Comment:
thought 2: do we know for sure that the remote will be "origin" for users? I can arbitrarily choose a name for the remote. If we're reasonably confident, that's fine!

This is another reason why you might want to notify the user what failed to be found and why.

Reply:
It's our only option, I think.